File: python/design-patterns/state/naive/player_naive.py
class Player:
    def __init__(self):
        self.is_playing = False
        self.is_paused  = False

    def play(self):
        if self.is_playing: return               # already playing
        self.is_paused = False
        self.is_playing = True                   # stopped/paused -> play

    def pause(self):
        if not self.is_playing: return           # ❌ can't pause if not playing
        self.is_playing = False
        self.is_paused = True

    def stop(self):
        # ❌ every method re-checks the same flags
        self.is_playing = False
        self.is_paused = False

File: cpp/design-patterns/state/naive/player_naive.cpp
class Player {
    bool isPlaying = false;
    bool isPaused  = false;
public:
    void play() {
        if (isPlaying) return;                   // already playing
        isPaused = false;
        isPlaying = true;                        // stopped/paused -> play
    }
    void pause() {
        if (!isPlaying) return;                  // ❌ can't pause if not playing
        isPlaying = false; isPaused = true;
    }
    void stop() {
        // ❌ every method re-checks the same flags
        isPlaying = false; isPaused = false;
    }
};